### PR TITLE
feat: Callbacks in `HudButtonComponent` constructor and `ViewportMargin` mixin to avoid code duplication

### PR DESCRIPTION
**Pull request by spydon**
# Description

Created a mixin to handle the margin to the viewport, mostly to remove the code duplication in `ButtonComponent` and `HudButtonComponent`, but it can also be useful in other cases when you don't want to wrap your component in `HudMarginComponent`.

Once the new camera system is in place this will of course have to be replaced.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame/lib/src/components/input/hud_button_component.dart [hunk 1]
```diff
@@ -36,52 +24,18 @@ class HudButtonComponent extends HudMarginComponent with Tappable {
     Iterable<Component>? children,
     int? priority,
   }) : super(
-          margin: margin,
+          button: button,
+          buttonDown: buttonDown,
           position: position,
+          onPressed: onPressed,
+          onReleased: onReleased,
```
Comment (wolfenrain) [inline packages/flame/lib/src/components/input/hud_button_component.dart:31]: Shouldnt this be part of a separate PR? For the sake of clear commits

**Review by erickzanardo — APPROVED** [review]
LGTM, just one small, optional comment.

**Review by wolfenrain — APPROVED** [review]
